### PR TITLE
Fix daemonization broken in #219

### PR DESCRIPTION
**Pull request by mariokostelac**

### CODE REVIEW

File: lib/shoryuken/cli.rb [hunk 1]
```diff
@@ -31,13 +31,13 @@ def run(args)
...
-      daemonize
-      write_pid
-
-      load_celluloid
+      daemonize(options)
+      write_pid(options)
 
       EnvironmentLoader.load(options)
 
+      load_celluloid
```
Comment (mariokostelac) [inline lib/shoryuken/cli.rb]: we have to load celluloid later so we could attach logger properly.

File: lib/shoryuken/cli.rb [hunk 2]
```diff
@@ -64,21 +64,27 @@ def run(args)
...
+      defined?(::Celluloid)
+    end
+
+    def daemonize(options)
+      return unless options[:daemon]
 
-      fail ArgumentError, "You really should set a logfile if you're going to daemonize" unless Shoryuken.options[:logfile]
+      fail ArgumentError, "You really should set a logfile if you're going to daemonize" unless options[:logfile]
+
+      if celluloid_loaded?
```
Comment (mariokostelac) [inline lib/shoryuken/cli.rb]: Daemonizing should fail if there is celluloid loaded, not loading celluloid, as we had before.

File: spec/shoryuken/cli_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+require 'shoryuken/cli'
+require 'shoryuken/launcher'
+
+RSpec.describe Shoryuken::CLI do
```
Comment (phstc) [inline spec/shoryuken/cli_spec.rb]: ❤️

**Comment by mariokostelac** [comment]
@phstc you good with this PR? :)

File: lib/shoryuken/cli.rb (hunk 2 above)
Comment (phstc) [inline lib/shoryuken/cli.rb]: @mariokostelac this is only to prevent users that manually `require celluloid` before requiring shoryuken?

Comment (mariokostelac) [inline lib/shoryuken/cli.rb]: yes, or us accidentally breaking it :)

File: lib/shoryuken/cli.rb [hunk 4]
```diff
@@ -64,21 +64,27 @@ def run(args)
     private
 
     def load_celluloid
-      raise "Celluloid cannot be required until here, or it will break Shoryuken's daemonization" if defined?(::Celluloid) && Shoryuken.options[:daemon]
```
Comment (mariokostelac) [inline lib/shoryuken/cli.rb]: @phstc it's just moved from here :)

**Comment by phstc** [comment]
@mariokostelac looks good to me :shipit:

**Comment by eugeneius** [comment]
Sorry for breaking this 😩 and thanks for fixing it @mariokostelac !

**Pull request merged**